tests/integration-tests/parsing/test_parsing_router.py

```
@@ -0,0 +1,216 @@
+        response = await client.post(
+            "/api/v1/parsing-status",
+            json={"repo_name": "nonexistent/owner-repo"},
+        )
+        assert response.status_code == 404
+
+
+class TestInputValidationEdgeCases:
+    """Test edge cases for input validation (Part 4.2 of plan)."""
+
+    async def test_post_parse_whitespace_only_repo_name(
+        self, client, test_user, mock_process_parsing_delay
```

Ruff (0.15.4)

[warning] Unused method argument: `test_user`

(ARG002)